docs/current_docs/quickstart/basics/basics.mdx

```diff
@@ -0,0 +1,288 @@
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./snippets/chaining/go/main.go
+```
+
+</TabItem>
+<TabItem value="Python">
+
+```python file=./snippets/chaining/python/__init__.py
+```
+
+</TabItem>
+<TabItem value="TypeScript">
+
+```typescript file=./snippets/chaining/typescript/index.ts
+```
+
+</TabItem>
+<TabItem value="PHP">
+
+```php file=./snippets/chaining/php/src/Basics.php
+```
+
+</TabItem>
+<TabItem value="Java">
+
+```java file=./snippets/chaining/java/src/main/java/io/dagger/modules/basics/Basics.java
+```
+
+</TabItem>
+<TabItem value="System shell">
+```shell
+# all equivalent
+dagger -c 'base | with-exec apk add bash git | publish ttl.sh/bar'
+dagger -c 'build | publish ttl.sh/bar'
+dagger -c build-and-publish
+```
+</TabItem>
+<TabItem value="Dagger Shell">
+```shell title="First type 'dagger' for interactive mode."
+# all equivalent
+base | with-exec apk add bash git | publish ttl.sh/bar
+build | publish ttl.sh/bar
+build-and-publish
+```
+</TabItem>
+</Tabs>
+:::
+
+## Using arguments and return values
+
+Dagger Functions are just like regular functions: they accept arguments and return values. In addition to common types (string, boolean, integer, arrays...), the Dagger API also defines powerful core types which Dagger Functions can use, such as `Directory`, `Container`, `Service`, `Secret`, and many more.
+
+Here's a revision of the previous example which splits it into two smaller functions: one to build the container, and one to publish it. The builder function accepts the container image string as an argument and returns a `Container` object. The publisher function accepts a `Container` as argument and returns the image identifier as a string.
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./snippets/arguments-return-values/go/main.go
+```
+
+</TabItem>
+<TabItem value="Python">
+
+FIXME
+
+</TabItem>
+<TabItem value="TypeScript">
+
+FIXME
+
+</TabItem>
+<TabItem value="PHP">
+
+FIXME
+
+</TabItem>
+<TabItem value="Java">
+
+FIXME
+
+</TabItem>
+</Tabs>
+
+[FIXME recording]
+
+:::tip FUNCTION NAMES
+When calling Dagger Functions, all names (functions, arguments, fields, etc.) are converted into a shell-friendly "kebab-case" style. This is why a Dagger Function named `FooPublish` in Go, `foo_publish` in Python and `fooPublish` in TypeScript/PHP/Java is called as `foo-publish` in Dagger Shell or on the command-line.
+:::
+
+## Speeding things up with caching
+
+One of Dagger's most powerful features is its built-in caching. This is typically most useful when dealing with package managers such as `apt`, `apk`, `npm`, `maven`, `pip` and similar. For these tools to cache properly, they need their own cache data (usually a directory) to be persisted between runs. Here's a Dagger Function that creates a cache volume to store the packages installed by __:
```

Comment:
This part confused me a bit because generally the **BIG** caching benefit is the "layer caching". The cache volumes are great too, but this reads like they're the only form of caching

Reply:
Modified to reflect both